packages/amplication-server/src/core/block/block.util.ts

```diff
@@ -40,3 +42,35 @@ export const mergeAllSettings = (
     ...mergedObj,
   };
 };
+
+export async function validateEntitlement(
+  blockType: EnumBlockType,
+  workspaceId: string
+): Promise<void> {
+  switch (blockType) {
+    case EnumBlockType.Module ||
+      EnumBlockType.ModuleAction ||
+      EnumBlockType.ModuleDto:
+      return await validateCustomActionsEntitlement(workspaceId);
+    default:
+      return;
+  }
+}
+
+async function validateCustomActionsEntitlement(
+  workspaceId: string
+): Promise<void> {
+  try {
+    const stiggClient = await this.getStiggClient();
```

Comment:
1. I am not sure where `this.getStiggClient()` is implemented, I believe this is a copy-paste from the billing service which actually has a `getStiggClient()` function. What am I missing?
2. Perhaps it's better not to have stigg in the block.util anyway, and instead use the relevant function from the billing service (and inject it here - e.g. `this.billingService.getBooleanEntitlement()`)

Reply:
@mulygottlieb  Logic moved to blockType service